src/webapp/context-providers/CurrentModuleContextProvider.tsx

```diff
@@ -0,0 +1,79 @@
+import { useEffect, useState, useMemo, useCallback } from "react";
+import { ModuleAccess } from "../../domain/entities/User";
+import { useHistory, useLocation } from "react-router-dom";
+import { defaultModuleContextState, CurrentModuleContext } from "../contexts/current-module-context";
+
+import { useAppContext } from "../contexts/app-context";
+
+export const CurrentModuleContextProvider: React.FC = ({ children }) => {
+    const history = useHistory();
+    const location = useLocation();
+    const queryParameters = useMemo(() => new URLSearchParams(location.search), [location]);
+    const moduleQueryParam = queryParameters.get("module");
+    const { currentUser } = useAppContext();
+
+    const [currentModuleAccess, setCurrentModuleAccess] = useState<ModuleAccess>(
+        defaultModuleContextState.currentModuleAccess
+    );
+
+    const [userModulesAccess, setUserModulesAccess] = useState<ModuleAccess[] | null>(null);
+
+    const changeCurrentModuleAccess = useCallback(
+        (updated: string) => {
+            const currentModuleAccess = userModulesAccess?.find(m => m.moduleName === updated);
+            if (currentModuleAccess) {
+                setCurrentModuleAccess(currentModuleAccess);
+                if (queryParameters.get("module")) {
+                    queryParameters.set("module", currentModuleAccess.moduleName);
+                    history.replace({ search: queryParameters.toString() });
+                }
+            }
+        },
+        [history, queryParameters, userModulesAccess]
+    );
+
+    const resetCurrentModuleAccess = () => {
+        setCurrentModuleAccess(defaultModuleContextState.currentModuleAccess);
+        queryParameters.delete("module");
+    };
+
+    useEffect(() => {
+        async function initModuleAccess() {
+            const { data: moduleAccessList } = await currentUser.userModulesAccess.runAsync();
```

Comment:
Same about `future.run`

Reply:
Addressed in https://github.com/EyeSeeTea/glass-dev/pull/35/commits/9e3f96184e88bbae9ca572610f2973d8fc45a599